automation_tools/__init__.py

```diff
@@ -1088,6 +1088,117 @@ def downstream_install(admin_password=None, run_katello_installer=True):
     else:
         return installer_options
 
+def repofile_install(admin_password=None, run_katello_installer=True, repo_url=None):
+    """Task to install Satellite 6.3 via repo files
+
+    The following environment variables affect this command:
+
+    ADMIN_PASSWORD
+        Optional, defaults to 'changeme'. Foreman admin password.
+    REPO_URL
+        URL for the compose repository file to fetch.
+    SATELLITE_VERSION
+        Satellite version.
+
+    """
+    # For Satellite 6.1 and 6.2 call downstream_install only.
+    if os.environ.get('SATELLITE_VERSION') != '6.3':
+       downstream_install()
+    if admin_password is None:
+        admin_password = os.environ.get('ADMIN_PASSWORD', 'changeme')
+
+    if repo_url is None:
+       repo_url = os.environ.get('REPO_URL')
+
+    run('yum install -y wget')
+    run('wget -O /etc/yum.repos.d/satellite63.repo {0}'.format(repo_url))
+
+    # Install required packages for the installation
+    run('yum install -y satellite')
+
+    installer_options = {
+        'foreman-admin-password': admin_password,
+    }
+    if run_katello_installer:
+        katello_installer(**installer_options)
+        # Ensure that the installer worked
+        run('hammer -u admin -p {0} ping'.format(admin_password))
+    else:
+        return installer_options
+
+def ak_install(admin_password=None, clean_beaker=True, run_katello_installer=True):
+    """Task to install Satellite 6.3 via Activation Keys
+
+    The following environment variables affect this command:
+
+    ADMIN_PASSWORD
+        Optional, defaults to 'changeme'. Foreman admin password.
+    SATELLITE_VERSION
+        Satellite version.
+    DOGFOOD_CERTURL
+        URL for the Candlepin Cert RPM.
```

Comment:
Since the candlepin cert rpm follows a standard naming, we could derive the full URL from the dog food system FQDN...